### PR TITLE
Added products view functionalities

### PR DESCRIPTION
**Pull request by heliomar-pena**
### Summary

- [Added queries for Products](https://github.com/Azordev/did-app/commit/cd29dd17e7362045bc95cf11d49792dab224768b)
- [Created Products Actions](https://github.com/Azordev/did-app/commit/0a54fb0f709f959c2ebab2dc118bc61cb8178727)
- [Created page pagination and search bar components](https://github.com/Azordev/did-app/commit/965f1e21df2f0df292dabf20fdf46477f9dbaf4c)
- [Created pagination utils](https://github.com/Azordev/did-app/commit/0334416d6cfbedd98634ed601088bb20784fa246)
- [Added ProductListItem component](https://github.com/Azordev/did-app/commit/792f6267f1b76bfe6b1b375c561b673f2f323ed1)
- [Added ProductList component](https://github.com/Azordev/did-app/commit/5667fadceaa8e5e817d6656f2cf700daaa82d0af)
- [Added products.scss and types to products](https://github.com/Azordev/did-app/commit/aa4b273b6a5fe1ad58535b6f12726c95e33ff227)
- [Created Products layout](https://github.com/Azordev/did-app/commit/897dfc6faad6463c588d4dc82c49c36f9e19d51c)
- [Created products logic](https://github.com/Azordev/did-app/commit/7d792e9456af560967dd575a2973899d40d4477e)
- [Adding order_by functionality](https://github.com/Azordev/did-app/commit/f37fe0cd628e6436e5443fd972af6a5b1b9c3954)

### Explanation

In more human words, I've created some functionalities for Products View in the Mobile APP, this includes:

- Show products by pagination
- Filter Products by Name, Description and Provider using Search Bar
- Order Products by price and alphabetic

### Screenshots

Normal view:
lastscreenshot

Loading view:
lastscreenshot

Filtered using search bar:
lastscreenshot

Sort by less price first:
lastscreenshot

### More details

There is some bugs that I'm working, for example:

1. When a user search the same word two times, loading state never end of load (and the network request never is done)

### Next steps

1. Fix bugs
2. Refactor a little the code -> Please help me to refactor leaving comments
3. Add a option to remove the "Order by" filter
4. Improve the "Clear filter" button logic
5. Create the Product detail view
6. Create the Providers view
7. Add functionality to filter Products by Provider selected

### CODE REVIEW

File: src/actions/ProductsActions.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  offset = 0,
+  query = '',
+  order_by = {
+    order_by_price: null,
+    order_by_name: null,
+  },
+}: getListOfProductsTypes) => {
+  query = `%${query}%`;
+
+  return new Promise<getListOfProductsReturnTypes>((res, rej) => {
```
Comment (Israel-Laguan) [inline src/actions/ProductsActions.ts]: Hmm maybe leave the full name of "res" and "rej", even if is more verbose

File: src/pages/Products/Products.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
+.ProductList {
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr;
+  gap: 20px;
+  margin: 20px;
+  margin: 20px;
```
Comment (Israel-Laguan) [inline src/pages/Products/Products.scss]: repeated style

File: src/actions/ProductsActions.ts (hunk 1 above)
Comment (Israel-Laguan) [reply src/actions/ProductsActions.ts]: Also in other files

File: src/pages/Products/Products.scss (hunk 2 above)
Comment (heliomar-pena) [reply src/pages/Products/Products.scss]: Thanks, I'll fix it

File: src/actions/ProductsActions.ts (hunk 1 above)
Comment (heliomar-pena) [reply src/actions/ProductsActions.ts]: Thanks! I'll fix it

**Review by e-azocar — APPROVED** [review]

**Review by Israel-Laguan — APPROVED** [review]

**Review by Israel-Laguan — APPROVED** [review]

**Comment by heliomar-pena** [comment]
I will merge to work in Providers view. Thank you for review!

**Pull request merged**